evm/src/fuzz/invariant/executor.rs

```diff
@@ -0,0 +1,420 @@
+                        calldata: calldata.clone(),
+                        gas: call_result.gas,
+                        stipend: call_result.stipend,
+                    });
+
+                    if !can_continue(
+                        &invariant_contract,
+                        call_result,
+                        &executor,
+                        &inputs,
+                        &mut failures.borrow_mut(),
+                        test_options,
+                    ) {
+                        break 'fuzz_run
+                    }
+
+                    // Generates the next call from the run using the recently updated
+                    // dictionary.
+                    inputs.extend(
+                        strat
+                            .new_tree(&mut branch_runner.borrow_mut())
+                            .map_err(|_| TestCaseError::Fail("Could not generate case".into()))?
+                            .current(),
+                    );
+                }
+
+                // We clear all the targeted contracts created during this run.
+                if !created_contracts.is_empty() {
+                    let mut writable_targeted = targeted_contracts.write();
+                    for addr in created_contracts.iter() {
+                        writable_targeted.remove(addr);
+                    }
+                }
+
+                fuzz_cases.borrow_mut().push(FuzzedCases::new(fuzz_runs));
+
+                Ok(())
+            });
+        }
+
+        let (reverts, invariants) = failures.into_inner().into_inner();
+
+        Ok(Some(InvariantFuzzTestResult { invariants, cases: fuzz_cases.into_inner(), reverts }))
+    }
+
+    /// Prepares certain structures to execute the invariant tests:
+    /// * Fuzz dictionary
+    /// * Targeted contracts
+    /// * Invariant Strategy
+    fn prepare_fuzzing(
+        &mut self,
+        invariant_contract: &InvariantContract,
+        test_options: InvariantTestOptions,
+    ) -> eyre::Result<InvariantPreparation> {
+        // Finds out the chosen deployed contracts and/or senders.
+        let (targeted_senders, targeted_contracts) =
+            self.select_contracts_and_senders(invariant_contract.address, invariant_contract.abi)?;
+
+        if targeted_contracts.is_empty() {
+            eyre::bail!("No contracts to fuzz.");
+        }
+
+        // Stores fuzz state for use with [fuzz_calldata_from_state].
+        let fuzz_state: EvmFuzzState = build_initial_state(&self.executor.backend().db);
+
+        // During execution, any newly created contract is added here and used through the rest of
+        // the fuzz run.
+        let targeted_contracts: FuzzRunIdentifiedContracts =
+            Arc::new(RwLock::new(targeted_contracts));
+
+        // Creates the invariant strategy.
+        let strat =
+            invariant_strat(fuzz_state.clone(), targeted_senders, targeted_contracts.clone())
+                .no_shrink()
```

Comment:
could you pls remind me why we do `no_shrink` here?

Reply:
still haven't explored it, that's why i added it to the follow up